parsers/UY.py

```diff
@@ -1,163 +1,224 @@
-        "datetime": obj["datetime"],
-        "production": dict(
-            [(k, obj[INV_MAP_GENERATION[k]]) for k in INV_MAP_GENERATION.keys()]
-        ),
-        "source": "ute.com.uy",
-    }
-
-    return data
+) -> list:
+    """collects production data from ADME and format all data points for target_datetime"""
+    if target_datetime is None:
+        target_datetime = arrow.utcnow().datetime
+    session = session or Session()
 
+    data = fetch_data(
+        zone_key=zone_key,
+        session=session,
+        target_datetime=target_datetime,
+        sheet_name="GPF",
+        logger=logger,
+    )
+    data = data.rename(columns=PRODUCTION_MODE_MAPPING)
+    data = data.groupby(data.columns, axis=1).sum()
+    production = data[[col for col in data.columns if col in PRODUCTION_MODES]]
+    production = pd.melt(production, var_name="production_mode", ignore_index=False)
+    all_data_points = []
+    for dt in production.index.unique():
+        production_dict = {}
+        data_dt = production.loc[production.index == dt]
+        for i in range(len(data_dt)):
+            row = data_dt.iloc[i]
+            mode = row["production_mode"]
+            if mode == "solar":
+                production_dict[mode] = fix_solar_production(dt=dt, row=row)
+            else:
+                production_dict[mode] = round(row.get("value"), 3)
+        data_point = {
+            "zoneKey": "UY",
+            "datetime": arrow.get(dt).datetime.replace(tzinfo=pytz.timezone(UY_TZ)),
+            "production": production_dict,
+            "source": "pronos.adme.com.uy",
+        }
+        all_data_points += [data_point]
+    return all_data_points
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_consumption(
+    zone_key: str = "UY",
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    """collects consumption data from ADME and format all data points for target_datetime"""
+    if target_datetime is None:
+        target_datetime = arrow.utcnow().datetime
+    session = session or Session()
 
+    data = fetch_data(
+        zone_key=zone_key,
+        session=session,
+        target_datetime=target_datetime,
+        sheet_name="GPF",
+        logger=logger,
+    )
+    data = data.rename(columns={"Demanda": "consumption"})
+
+    consumption = data[["consumption"]].to_dict(orient="index")
+    all_data_points = []
+    for dt in consumption:
+        data_point = {
+            "zoneKey": "UY",
+            "datetime": arrow.get(dt).datetime.replace(tzinfo=pytz.timezone(UY_TZ)),
+            "consumption": round(consumption[dt]["consumption"], 3),
+            "source": "pronos.adme.com.uy",
+        }
+        all_data_points += [data_point]
+    return all_data_points
+
+
+@refetch_frequency(timedelta(days=1))
 def fetch_exchange(
-    zone_key1: str = "UY",
-    zone_key2: str = "BR-S",
+    zone_key1: str,
```

Comment:
If we're fetching some exchanges with different parsers I'd suggest asserting that the given keys don't correspond to the exchange we fetch otherwise, or at least log it